ui/src/app/analytics/analytics.service.ts

```diff
@@ -1,39 +1,115 @@
 import { Injectable } from '@angular/core';
-import { Http, Response } from '@angular/http';
+import { Http, Response, RequestOptionsArgs } from '@angular/http';
+
 import { Observable } from 'rxjs/Observable';
+import { Subscription } from 'rxjs/Subscription';
 import 'rxjs/add/operator/catch';
 import 'rxjs/add/operator/map';
 
+import { ErrorHandler, Page } from '../shared/model';
+import { Counter } from './counters/model/counter.model';
+import { HttpUtils } from '../shared/support/http.utils';
+
+/**
+ * @author Gunnar Hillert
+ */
 @Injectable()
 export class AnalyticsService {
 
-  private appstUrl = '/apps';
+  private metricsCountersUrl = '/metrics/counters';
+
+  public counters: Page<Counter>;
+  public _counterInterval = 2;
+  public counterPoller: Subscription;
+  public countersCache: number[];
 
-  constructor(private http: Http) { }
+  constructor(private http: Http, private errorHandler: ErrorHandler) {
+  }
 
-  getApps(): Observable<any[]> {
-    return this.http.get(this.appstUrl)
-                    .map(this.extractData)
-                    .catch(this.handleError);
+  set counterIntervall(rate: number) {
+    if (rate && !isNaN(rate)) {
+      if (rate < 0.01) {
+          rate = 0;
+          this.stopPollingForCounters();
+      } else {
+        console.log('Setting intervall to ' + rate);
+        this._counterInterval = rate;
+        if (this.counterPoller && !this.counterPoller.closed) {
+          this.stopPollingForCounters();
+          this.startPollingForCounters();
+        }
+      }
+    }
   }
 
-  private extractData(res: Response) {
-    const body = res.json();
-    return body._embedded;
+  get counterIntervall(): number {
+    return this._counterInterval;
   }
 
-  private handleError (error: Response | any) {
-    // In a real world app, you might use a remote logging infrastructure
-    let errMsg: string;
-    if (error instanceof Response) {
-      const body = error.json() || '';
-      const err = body.error || JSON.stringify(body);
-      errMsg = `${error.status} - ${error.statusText || ''} ${err}`;
-    } else {
-      errMsg = error.message ? error.message : error.toString();
+  public totalCacheSize() {
+    return Math.max(Math.ceil(60 / this._counterInterval), 20);
+  }
+
+  startPollingForCounters() {
+    if (!this.counterPoller || this.counterPoller.closed) {
+      this.counterPoller = Observable.interval(this._counterInterval * 1000)
+        .switchMap(() => this.getAllCounters(true)).subscribe();
```

Comment:
If you don't have redis setup, backend service errors
```
$ curl "http://localhost:9393/metrics/counters?page=0&size=10&detailed=true"
[{"logref":"RedisConnectionFailureException","message":"Cannot get Jedis connection; nested exception is redis.clients.jedis.exceptions.JedisConnectionException: Could not get a resource from the pool","links":[]}]
```
I think this is causing whole Observable to fail and no further http requests are made.

Reply:
@jvalkeal Can you clarify the test case.  I killed my instance of redis,  refreshed the page and the counters disappeared.  Once I started the redis instance , and refreshed the page the counters were back..

Reply:
We also need to add the following to the subscribe()
```
    error => {
        this.toastyService.error(error);
      }
```

Reply:
Created a backend story for this also: https://github.com/spring-cloud/spring-cloud-dataflow/issues/1636